sound/soc/sof/intel/hda-loader.c

```diff
@@ -336,8 +336,11 @@ int hda_dsp_cl_boot_firmware_iccmax(struct snd_sof_dev *sdev)
 
 	stripped_firmware.size = plat_data->fw->size - plat_data->fw_offset;
 
-	/* prepare capture stream for ICCMAX */
-	iccmax_stream = hda_cl_stream_prepare(sdev, HDA_CL_STREAM_FORMAT, stripped_firmware.size,
+	/*
+	 * Prepare capture stream for ICCMAX. We do not need to store
+	 * the data, so use a buffer of PAGE_SIZE for receiving.
+	 */
+	iccmax_stream = hda_cl_stream_prepare(sdev, HDA_CL_STREAM_FORMAT, PAGE_SIZE,
```

Comment:
how does this solve an out of memory issue? It just reduces the amount of memory used, but not why we keep allocating it. it feels like this makes the problem less likely to be seen but doesn't address the problem.

What am I missing?

Reply:
@plbossart This PR fixes a bug in the code. The iccmax flow requires host to set up a capture DMA stream. There is no data transferred AFAIK, but to safely program and start the host side DMA, we need to have SOME memory configured to the DMA hardware. I chose PAGE_SIZE as the most convenient small size. If anyone has better suggestions, I'm all ears.

This change has the side-effect that it decreases pressure on the memory subsystem on systems where memory is really running low (like in the linked bug). With the IMR boot enabled, we don't need to allocate big chunks of DMA-capable memory at each resume. But with the iccmax stream bug in place, we lose this advantage and may hit this problem on systems with low and fragmented free memory.

I'm not claiming this fixes the bug completely, but it certainly was found while debugging the bug, and this fix reduces pressure on the memory subsystem.

Reply:
Thanks @kv2019i

My point is that this feels like a first step but additional work could be done, such as

a)  pre-allocate this page upfront? It's 4KB so unlikely to ever impact other usages.
b) re-inspect if/why we need this iccmax stream when we use IMR.

Reply:
For (a), we only need this very briefly, so why hold on to valuable RAM if we don't need it? OTOH, if not even a single page can be allocated at system resume, I think it's a lost case anyways.

For (b), @ranj063 is checking this. My take this is still needed as this is part of the power-up process (and IMR-boot doesn't affect this part).

Reply:
@ranj063 Did you have a chance to investigate this?

Reply:
Conclusion is that the capture stream is also for imrboot.